example/ext-pipe-plugin-example/src/main/java/org/apache/iotdb/extpipe/ExtPipeSinkWriterImpl.java

```diff
@@ -49,88 +49,85 @@ public void open() {
   }
 
   @Override
-  public synchronized void insertBoolean(String[] path, long timestamp, boolean value)
+  public void insertBoolean(String sgName, String[] path, long timestamp, boolean value)
       throws IOException {
     //== Here, handle inserted Boolean type data from IoTDB.
     //extSession.insertBoolean(...);
     //...
   }
 
   @Override
-  public synchronized void insertInt32(String[] path, long timestamp, int value)
+  public void insertInt32(String sgName, String[] path, long timestamp, int value)
       throws IOException {
     //== Here, handle inserted Int32 type data from IoTDB.
     //extSession.insertInt32(...);
     //...
   }
 
   @Override
-  public synchronized void insertInt64(String[] path, long timestamp, long value)
-      throws IOException {
+  public void insertInt64(String sgName, String[] path, long time, long value) throws IOException {
     //== Here, handle inserted Int64 type data from IoTDB.
     //...
   }
 
   @Override
-  public synchronized void insertFloat(String[] path, long timestamp, float value)
-      throws IOException {
+  public void insertFloat(String sgName, String[] path, long time, float value) throws IOException {
     //== Here, handle inserted float type data from IoTDB.
     //extSession.insertFloat(...);
     //...
   }
 
   @Override
-  public synchronized void insertDouble(String[] path, long timestamp, double value)
-      throws IOException {
+  public void insertDouble(String sgName, String[] path, long time, double value) throws IOException {
     //== Here, handle inserted double type data from IoTDB.
     //extSession.insertDouble(...);
     //...
   }
 
   @Override
-  public synchronized void insertText(String[] path, long timestamp, String value)
-      throws IOException {
+  public void insertText(String sgName, String[] path, long time, String value) throws IOException {
     //== Here, handle inserted Text type data from IoTDB.
     //extSession.insertText(...);
     //..
   }
 
-  @Override
-  public synchronized void insertVector(String[] path, DataType[] dataTypes, long timestamp,
-                                        Object[] values)
-      throws IOException {
-    //== Here, handle inserted Vector type data from IoTDB.
-    //extSession.insertVector(...);
-    //...
-  }
+//  @Override
+//  public void insertVector(String[] path, DataType[] dataTypes, long timestamp,
+//                                        Object[] values)
+//      throws IOException {
+//    //== Here, handle inserted Vector type data from IoTDB.
+//    //extSession.insertVector(...);
+//    //...
+//  }
```

Comment:
Uncomment

Reply:
Thank @Cpaulyz 
In current design, this interface method will not be used.
So I have just removed this part.